makefile

```diff
@@ -23,23 +23,27 @@ filenamesc := ft_strlen.c \
 			ft_strncmp.c \
 			ft_atoi.c \
 
-filenameso := $(filenamesc:%.c=%.o) 
-CC = gcc -Wall -Wextra -Werror
+filenameso := $(filenamesc:%.c=%.o)
+
+CC = gcc
+CFLAGS = -Wall -Wextra -Werror
+NAME = libft.a
 
 define dotocreator =
 $(firstword $^): $(firstword $^).c
-$(CC) -c $(firstword $^).c
+$(CC) $(CFLAGS) -c $(firstword $^).c
 endef
 
 #===========#
 #Compilation#
 #===========#
 
-test: test.c libft.a libft.h
-	$(CC) -o test libft.a test.c
+test: test.c $(NAME) libft.h
+	$(CC) -o test $(NAME) test.c
 
-libft.a: $(filenameso) 
-	ar -rc libft.a $(filenameso)
+$(NAME): $(filenameso) 
+	ar -rc $(NAME) $(filenameso)
+	ranlib $(NAME)
```

Comment:
It's supposed to index your library to make it faster.
As you can see, I don't know too much about the subject.